### ISSUE TITLE
Discrepancy in name of top node vs name of graph from graphs exported from pre-4.3

### ISSUE THREAD

**Issue by adamlodge** [archesproject/arches#4167]
In 4.3, the name of the top node is by default consistent with the name of the graph that it (the node) is the top node of. ... which is a good thing.  However, graphs imported via json from older versions in Arches often have graphs where the top node has a different name than the graph itself.  

Anyhow, this discrepancy can cause some confusion for (human) users because there is no place in the 4.3 UI to differentiate between top nodes and graph names, but also in read-only places in the UI, the value is drawn from the two different places.

Bottom line:  I wrote this sql to resolve the discrepancy.  If it passes muster, would be nice to use it as a basis for a post-import check when importing json graphs in 4.3.

This assumes that you want to keep the name specified as the graph name over the name specified as the top node name.

UPDATE nodes a
SET name = b.name
FROM graphs b
WHERE 1=1
AND a.graphid = b.graphid
AND a.istopnode = true
AND a.name <> b.name;

**Comment by adamlodge** [comment]
@mradamcox Is this an issue you have had to wrestle with in your system earlier Arches to later Arches migrations?

**Comment by mradamcox** [comment]
@adamlodge I haven't really moved any version from pre-4.3 to 4.3 so can't really say. You summary does make sense though, and I agree it's better to have the top node tied to the name of the resource model.

**Comment by adamlodge** [comment]
@mradamcox .. When the time comes, if this solution is not implemented as part an automagic import routine, at least it's here for you to use directly in the database if you need.

**Comment by chiatt** [comment]
Closing due to inactivity
